static/js/settings_account.js

```diff
@@ -254,68 +255,68 @@ exports.set_up = function () {
-            $("#get_api_key_box").show();
-        } else {
+    var setup_api_key_modal = _.once(function () {
+        $('.account-settings-form').append(render_settings_api_key_modal());
+        $("#api_key_value").text("");
+        $("#show_api_key").hide();
+
+        if (page_params.realm_password_auth_enabled === false) {
             // Skip the password prompt step, since the user doesn't have one.
             $("#get_api_key_button").click();
         }
-        $("#api_key_button_box").hide();
-    });
 
-    $("#get_api_key_box").hide();
-    $("#show_api_key_box").hide();
+        $("#get_api_key_button").on("click", function (e) {
+            var data = {};
+            e.preventDefault();
+            e.stopPropagation();
 
-    $("#get_api_key_button").on("click", function (e) {
-        var data = {};
-        e.preventDefault();
-        e.stopPropagation();
+            data.password = $("#get_api_key_password").val();
+            channel.post({
+                url: '/json/fetch_api_key',
+                dataType: 'json',
+                data: data,
+                success: function (data) {
+                    $("#get_api_key_password").val("");
+                    $("#api_key_value").text(data.api_key);
+                    // The display property on the error bar is set to important
+                    // so instead of making display: none !important we just
+                    // remove it.
+                    $('#api_key_status').remove();
+                    $("#password_confirmation").hide();
+                    $("#show_api_key").show();
+                },
+                error: function (xhr) {
+                    ui_report.error(i18n.t("Error"), xhr, $('#api_key_status').expectOne());
+                    $("#show_api_key").hide();
+                    $("#api_key_modal").show();
+                },
+            });
+        });
 
-        data.password = $("#get_api_key_password").val();
-        channel.post({
-            url: '/json/fetch_api_key',
-            dataType: 'json',
-            data: data,
-            success: function (data) {
-                var settings_status = $('#account-settings-status').expectOne();
-
-                $("#get_api_key_password").val("");
-                $("#api_key_value").text(data.api_key);
-                $("#show_api_key_box").show();
-                $("#get_api_key_box").hide();
-                settings_status.hide();
-            },
-            error: function (xhr) {
-                ui_report.error(i18n.t("Error getting API key"), xhr, $('#account-settings-status').expectOne());
-                $("#show_api_key_box").hide();
-                $("#get_api_key_box").show();
-            },
+        $("#show_api_key").on("click", "button.regenerate_api_key", function () {
+            channel.post({
+                url: '/json/users/me/api_key/regenerate',
+                idempotent: true,
```

Comment:
reason for using this isn't clear to me

Reply:
With idempotent set to true, `channel.post` will retry the request if it fails the first time. Do we want that here?

Reply:
No; we should only set that for things with no side effects for being done twice, which isn't the case here.